jetty-ee10/jetty-ee10-proxy/src/main/java/org/eclipse/jetty/ee10/proxy/AsyncMiddleManServlet.java

```diff
@@ -777,7 +777,7 @@ public static class GZIPContentTransformer implements ContentTransformer
 
         private final List<ByteBuffer> buffers = new ArrayList<>(2);
         private final ContentTransformer transformer;
-        private final ContentDecoder decoder;
+        private final GZIPContentDecoder decoder;
```

Comment:
You should set the field type as `GZIPDecoder` to make it explicit that using this subclass is mandatory.

Reply:
It is an internal implementation detail, like declaring `Map` but then using `HashMap`.
The key point here is that we could get rid of the `jetty-client`'s `GZIPContentDecoder` and we can now use `jetty-http`s version without problems.